api/metrics/views.py

```diff
@@ -0,0 +1,196 @@
+from datetime import timedelta, datetime
+
+import pytz
+from django.utils import timezone
+from django.http import JsonResponse
+from rest_framework.exceptions import ValidationError
+from rest_framework import permissions as drf_permissions
+from elasticsearch.exceptions import NotFoundError, RequestError
+
+from osf.models import Preprint
+from framework.auth.oauth_scopes import CoreScopes
+from api.base.permissions import TokenHasScope
+from osf.metrics import PreprintDownload, PreprintView
+from api.metrics.permissions import IsPreprintMetricsUser
+from api.metrics.serializers import PreprintMetricSerializer
+from api.base.views import JSONAPIBaseView
+
+
+class PreprintMetricMixin(JSONAPIBaseView):
+    permission_classes = (
+        drf_permissions.IsAuthenticated,
+        drf_permissions.IsAdminUser,
+        IsPreprintMetricsUser,
+        TokenHasScope,
+    )
+
+    required_read_scopes = [CoreScopes.METRICS_READ]
+    required_write_scopes = [CoreScopes.METRICS_WRITE]
+
+    serializer_class = PreprintMetricSerializer
+
+    @property
+    def metric_type(self):
+        raise NotImplementedError
+
+    @property
+    def metric(self):
+        raise NotImplementedError
+
+    def add_search(self, search, query_params, **kwargs):
+        """
+        get list of guids from the kwargs
+        use that in a query to narrow down metrics results
+        """
+        preprint_guid_string = query_params.get('guids')
+        if not preprint_guid_string:
+            raise ValidationError('You must provide one or more preprint guids to gather metrics for')
+        preprint_guids = preprint_guid_string.split(',')
+        for guid in preprint_guids:
+            preprint = Preprint.load(guid)
```

Comment:
Do we think this is necessary? They're loaded literally just to make sure it's a real preprint, and nothing else is done with it. Might just be slowing things down for no reason, if it's not a preprint guid it won't return results anyhow.